core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2051,6 +2051,14 @@ package object config {
       .timeConf(TimeUnit.MILLISECONDS)
       .createOptional
 
+  private[spark] val BIN_PACK_ENABLED =
+    ConfigBuilder("spark.scheduler.binPack.enabled")
```

Comment:
maybe defining `spark.scheduler.taskAssignmentStrategy` makes it more extendable?